### PR TITLE
[bug] Small correction on waitForBuildCompleted if buildconfig not yet existing 

### PR DESCRIPTION
**Pull request by radtriste**
issue #305

Wait for latest build to exist

### CODE REVIEW

File: core/src/main/java/cz/xtf/core/openshift/OpenShiftWaiters.java [hunk 1]
```diff
@@ -21,6 +21,7 @@
 	}
 
 	public Waiter hasBuildCompleted(String buildConfigName) {
+		isLatestBuildPresent(buildConfigName).waitFor();
```
Comment (Crumby) [inline core/src/main/java/cz/xtf/core/openshift/OpenShiftWaiters.java]: Actual waiting goes against convention of this class. None of the methods waits for anything, they only return executable Waiter objects.

Comment (radtriste) [reply core/src/main/java/cz/xtf/core/openshift/OpenShiftWaiters.java]: ok.. Need to set the documentation then.

Comment (radtriste) [reply core/src/main/java/cz/xtf/core/openshift/OpenShiftWaiters.java]: I will update this

Comment (Crumby) [reply core/src/main/java/cz/xtf/core/openshift/OpenShiftWaiters.java]: Updating the docs is good but you can also fix the method so the user doesn't have to call `isLatestBuildPresent` and then this one by rewriting the Waiter implementation for this method. It'll be combination of the one isLatestBuildPresent and the one that it calls right now.

Comment (radtriste) [reply core/src/main/java/cz/xtf/core/openshift/OpenShiftWaiters.java]: ok. Did a simple implementation using Optional and returning null string if not yet existing.

Could also creates a SimpleWaiter with the condition with condition not null and Phase Complete or Failed but thought it clearer and in sync with the other 'hasBuildCompleted' method. Let me know what you think of it.

**Comment by radtriste** [comment]
squash and rebase done.

**Pull request merged**